openverse_catalog/dags/common/loader/s3.py

```diff
@@ -28,6 +35,34 @@ def copy_file_to_s3_staging(
     s3.load_file(tsv_file_path, staging_key, bucket_name=s3_bucket)
 
 
+def copy_file_to_s3(
+    tsv_file_path,
+    s3_bucket,
+    s3_prefix,
+    aws_conn_id,
+    ti,
+):
+    """
+    Copy a TSV file to S3 with the given prefix.
+    The TSV's version is pushed to the `tsv_version` XCom, and the constructed
+    S3 key is pushed to the `s3_key` XCom.
+    The TSV is removed after the upload is complete.
+    """
+    if tsv_file_path is None:
+        raise FileNotFoundError("No TSV file path was provided")
+    tsv_file = Path(tsv_file_path)
+    if not tsv_file.exists():
+        raise AirflowSkipException(f"TSV file {tsv_file} does not exist.")
+    tsv_version = paths.get_tsv_version(tsv_file_path)
+    s3_key = f"{s3_prefix}/{tsv_file.name}"
+    logger.info(f"Uploading {tsv_file_path} to {s3_bucket}:{s3_key}")
+    s3 = S3Hook(aws_conn_id=aws_conn_id)
+    s3.load_file(tsv_file_path, s3_key, bucket_name=s3_bucket)
```

Comment:
Is it possible that there can be a file with the same name already placed in the bucket? I'm wondering if worth the chance to run some validation to make sure if there's a file with the same name.

[As per the source code](https://github.com/apache/airflow/blob/4dd751d2fc3f091417146079d7d1b3ac6dd86c9c/airflow/providers/amazon/aws/hooks/s3.py#L497) the default behavior is to overwrite the file, is this the expected behavior?

Reply:
Good question! The S3 key is constructed using the provider, media type, TSV version, and timestamp for the run:
https://github.com/WordPress/openverse-catalog/blob/df7f68460a7ceb14cc39a02743ba8190411c017c/openverse_catalog/dags/common/storage/media.py#L175

The only case where we'll have a duplicate file is if the TSV is attempted to be uploaded after it has already been successfully uploaded. This would require some pretty intentional manual intervention to accomplish, so I think we're good :slightly_smiling_face: 

Also it looks like the default is actually to raise an error if the file exists, so we won't perform any unintentional overwrites either:
https://github.com/apache/airflow/blob/4dd751d2fc3f091417146079d7d1b3ac6dd86c9c/airflow/providers/amazon/aws/hooks/s3.py#L483